src/Microsoft.AspNetCore.Mvc.TagHelpers/PartialTagHelper.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    /// <summary>
+    /// Renders a partial view.
+    /// </summary>
+    [HtmlTargetElement("partial", Attributes = "name", TagStructure = TagStructure.WithoutEndTag)]
+    public class PartialTagHelper : TagHelper
+    {
+        private const string ForAttributeName = "asp-for";
+        private readonly ICompositeViewEngine _viewEngine;
+        private readonly IViewBufferScope _viewBufferScope;
+
+        public PartialTagHelper(
+            ICompositeViewEngine viewEngine,
+            IViewBufferScope viewBufferScope)
+        {
+            _viewEngine = viewEngine ?? throw new ArgumentNullException(nameof(viewEngine));
+            _viewBufferScope = viewBufferScope ?? throw new ArgumentNullException(nameof(viewBufferScope));
+        }
+
+        /// <summary>
+        /// The name of the partial view used to create the HTML markup.
```

Comment:
Does this do just view "names" (e.g. `_LoginPartial`) or does it also do paths? Do we need to be clear about this in the docs?

Reply:
I didn't see any tests using a path, but would it work? And if paths aren't supported:
1. Should they be?
2. The docs need to be clearer about what a "partial view name" is (perhaps by giving an example)

Reply:
Looks like these properties are handled identically to e.g. `viewName` parameters in `Controller.View(...)` methods. Those are generally documented as `The name of the view that is rendered to the response.`. Same for the corresponding result properties e.g. `ViewResult.ViewName`.

If we need more documentation about what names, paths, et cetera mean, suggest we handle that in a separate issue. For now, let's be consistent.

Reply:
It does path - https://github.com/aspnet/Mvc/pull/7089/files#diff-883159b6acb104f0fb64afa28c27a9ccR2 . The name I think comes from the IHtmlHelper docs. I'd be happy to update it if you have suggestions

Reply:
Oh, and "partial view name" doesn't appear anywhere. Isn't "name of the partial view" clear enough (assuming we stick with just "name" for now)?

Reply:
> For now, let's be consistent.

I'd rather have correct information than have consistent-but-useless information

> Oh, and "partial view name" doesn't appear anywhere.

These mean the same thing so I don't think this is critical.

...

So, let's come up with some good wording here and then apply elsewhere.

Reply:
How 'bout "The name or path of the [partial] view that is rendered to the response." We do lots of shenanigans under the covers handling absolute and relative paths as well as distinguishing paths from names. But, describing exactly how we choose to interpret "name" versus "path" doesn't help much -- especially because it's specific to the `RazorViewEngine`.

I'm not positive what extra words we need to inform readers what distinguishes a "partial" view. It's a view found with `isMainPage: false` but the interfaces say only that `isMainPage` "Determines if the page being found is the main page for an action.". The details are again specific to the `IViewEngine` implementation. (For `RazorViewEngine`, this parameter mainly controls whether or not `_ViewStart.cshtml` files are used.)